src/sophys/common/plans/parallel_wrapper.py

```diff
@@ -0,0 +1,44 @@
+from threading import Event, Thread
+
+
+def yield_msg(msg):
+    yield msg
+
+def all_plans_have_finished(plan_has_finished_list):
+    for plan_finished in plan_has_finished_list.values():
+        if not plan_finished.is_set():
+            return True
+    return False
+    
+def parallel_plans_wrapper(*args):
+    parallel_plan_list = list(args)
+    plan_has_finished_list = {}
+    current_event = {}
+    current_thread = {}
+
+    # Initialize dictionaries to store the thread, event and has_finished data
+    for list_id, _ in enumerate(parallel_plan_list):
+        plan_has_finished_list[list_id] = Event()
+        current_event[list_id] = parallel_plan_list[list_id]
```

Comment:
I think `current_event[list_id]` shouldn't be set here, because its value is never used, since `current_event[list_id] = next(parallel_plan_list[list_id])` overwrites it.